### ISSUE TITLE
Single quote inside heredoc causes remainder of file to be rendered as string

### ISSUE THREAD

**Issue by cdown** [atom/language-shellscript#60]
Originally from https://github.com/github/linguist/issues/3290:

Currently this is still considered a string opening quote, even though it doesn't act as one

``` bash
cat << EOF
test ' test
EOF
echo foo bar
# Note that these are still highlighted as if they were a string
```

I'll take a look at fixing this when/if I have time, but otherwise I think this is worth fixing anyway.

**Comment by 50Wliu** [comment]
Blocked on https://github.com/atom/first-mate/issues/83.

**Comment by nickooolas** [comment]
Is this related to my situation too? 

![image](https://cloud.githubusercontent.com/assets/12520584/23199511/ef50f700-f922-11e6-83f7-e733edb1c315.png)

**Comment by nickooolas** [comment]
This is related to my current issue in right?

![image](https://cloud.githubusercontent.com/assets/12520584/23199572/43efb774-f923-11e6-998e-83d625b41fbd.png)

**Comment by 50Wliu** [comment]
No, you don't have any heredocs.  Please open a new issue if one doesn't already exist.

**Comment by franklinyu** [comment]
Guess [this file](https://github.com/vishnubob/wait-for-it/blob/ac372412a1bb2b96606e2993a3c30589d131569d/wait-for-it.sh) is also affected by this issue. If I remove [the space in `<< USAGE`](https://github.com/vishnubob/wait-for-it/blob/ac372412a1bb2b96606e2993a3c30589d131569d/wait-for-it.sh#L10) (making it `<<USAGE`) then it works fine.

**Comment by 50Wliu** [comment]
Fixed by #77.
